oop-relationships/practice-composition.md

```diff
@@ -0,0 +1,7 @@
+# Practice: Composition
+
+## Directions
+
+Complete all questions below.
+
+## Practice
```

Comment:
Here's what I've got so far:

<!-- prettier-ignore-start -->
### !challenge
* type: checkbox
* id: 31f651e3-aa01-47f4-8e02-ece744f35515
* title: Practice Composition I

##### !question
'AdaBank' is a mobile app for users to keep track of their banking accounts. An `Account_Summary` will often display balances from a user's `Savings_Account` and `Credit_Card_Account`. Select the option(s) that best describes this OOP relationship. 
##### !end-question

##### !options
* `Account_Summary` is the parent class, while `Savings Account` and `Credit Card Account` are child classes.
* `Account_Summary` **has-a** relationship with `Savings_Account` and `Credit_Card_Account`.
* `Savings_Account` and `Credit_Card_Account` are component classes while `Account_Summary` is the composite class. 
##### !end-options

##### !answer
* `Account_Summary` **has-a** relationship with `Savings_Account` and `Credit_Card_Account`.
* `Savings_Account` and `Credit_Card_Account` are component classes while `Account_Summary` is the composite class. 
##### !end-answer
### !end-challenge
<!--prettier-ignore-end -->

<!-- prettier-ignore-start -->
### !challenge
* type: multiple-choice
* id: 6770bb4d-c21e-4505-ad5e-aed92cb70a5f
* title: Practice Composition II

##### !question
Willow is creating a Timesheet app that will display an employee's biweekly hours and pay. Provided this code snippet, which option describes the relationship between the `Pay_Roll` and `Employee` class?

``` Python
class Employee:
  def __init__(self, name, pay_rate, hours):
    self.name = name
    self.pay_rate = pay_rate
    self.hours = hours 
    self.payroll_summary = PayRoll(self.pay_rate, self.hours)

  def display_payroll_summary(self):
    return "Total Hours: " + str(self.hours) + "\nTotal Pay: " + "${:,}".format(self.payroll_summary.annual_rate()) )

class PayRoll:
  def __init__(self, pay, hours):
    self.pay = pay
    self.hours = hours
    
  def annual_rate(self):
    return self.pay* 12* self.hours

obj_emp = Employee("Sally", 30, 200)
print(obj_emp.display_payroll_summary())
```

##### !end-question

##### !options
* `Pay_Rate` is the child class, `Employee` is the parent class.
* `Pay_Rate` is the composite  class, `Employee` is the component class
* `Pay_Rate` is the component class, `Employee` is the composite class
##### !end-options

##### !answer
* `Pay_Rate` is the component class, `Employee` is the composite class
##### !end-answer
### !end-challenge
<!-- prettier-ignore-end -->